File: app/src/main/java/com/amirreza/domain/entity/CityEntity.java
package com.amirreza.domain.entity;


public class CityEntity {
    private double lat;
    private double lon;
    private String city;
    private String country;

    public CityEntity(double lat, double lon, String city, String country) {
        this.lat = lat;
        this.lon = lon;
        this.city = city;
        this.country = country;
    }
    public CityEntity(String lat, String lon) {
        this.lat = Double.parseDouble(lat);
        this.lon = Double.parseDouble(lon);
    }

    public double getLat() {
        return lat;
    }

    public double getLon() {
        return lon;
    }

    public String getCity() {
        return city;
    }

    public String getCountry() {
        return country;
    }
}

File: app/src/main/java/com/amirreza/domain/entity/CityEntity.kt
package com.amirreza.domain.entity

class CityEntity(lat: String, lon: String) {
    var lat: Double
        private set
    var lon: Double
        private set
    var city: String? = null
        private set
    var country: String? = null
        private set

    init {
        this.lat = lat.toDouble()
        this.lon = lon.toDouble()
    }
}